MyMethods.cs

```diff
@@ -0,0 +1,257 @@
+            catch (Exception e)
+            {
+                con.Error = true;
+                con.Description = e.Message;
+            }
+            return null;
+        }
+        public static string GetText(IWebElement el)
+        {
+            if (el != null)
+            {
+                return el.Text;
+            }
+            else
+            {
+                return " - ";
+            }
+        }
+        public static void ExtractDataFromGoogle(ref IWebDriver driver, ref string[] lines, ref Worksheet ws, Control con, LogWriter log)
+        {
+            IWebElement SearchBox;
+            IWebElement Temperatura;
+            IWebElement Opady;
+            IWebElement Wiatr;
+
+            int LastRow = ws.UsedRange.Rows.Count;
+            int row = LastRow +1;
+            int counter = 0;
+            foreach (string line in lines)
+            {
+                con.Error = false;
+                con.Description = "";
+                Console.WriteLine("City: " + line);
+                driver.Navigate().GoToUrl("https://www.google.pl/");
+                SearchBox = MyMethods.findElementAfterPreparingPage(driver, By.Id("lst-ib"), ref con);
+                SearchBox.Click();
+                SearchBox.SendKeys(line + " pogoda" + OpenQA.Selenium.Keys.Enter);
+                Temperatura = MyMethods.findElementAfterPreparingPage(driver, By.Id("wob_tm"), ref con);
+                Opady = MyMethods.findElementAfterPreparingPage(driver, By.Id("wob_pp"), ref con);
+                Wiatr = MyMethods.findElementAfterPreparingPage(driver, By.Id("wob_ws"), ref con);
+
+                DateTime now = DateTime.Now;
+
+                ws.Cells[row, 1].Value = String.Format("{0:yyyy/MM/dd HH:mm:ss}", now);
+                ws.Cells[row, 2].Value = line;
+                ws.Cells[row, 3].Value = MyMethods.GetText(Temperatura);
+                ws.Cells[row, 4].Value = MyMethods.GetText(Opady);
+                ws.Cells[row, 5].Value = MyMethods.GetText(Wiatr);
+                if (con.Error == true)
+                {
+                    ws.Cells[row, 6].Value = "Fail";
+                    ws.Rows[row].Interior.Color = 5296274;
+                }
+                else
+                {
+                    ws.Cells[row, 6].Value = "Success";
+                }
+                log.LogWrite(con, "Extract data for " + line);
+                
+                row++;
+                counter++;
+            }
+        }
+        public static string CheckStatus(Control con, string s)
+        {
+            if (con.Error == true)
+            {
+                Console.WriteLine(s + ". Action failed.\n" + con.Description);
+                return s + ". Action failed.\n" + con.Description;
+            }
+            else
+            {
+                Console.WriteLine(s + ". Action successful.");
+                return s + ". Action successful.";
+            }
+
+        }
+        
+}
+    public class LogWriter

```

Comment:
Podzielmy to na dwie klasy klasa, nazwijmy ja:
Logger - klasa abstracyjna
FileLogger - klasa rozsz. klase Loggera